MASH-FRET/source/simulations/phtn2arb.m

```diff
@@ -1,25 +1,12 @@
-function dat = phtn2arb(varargin)
+function dat = phtn2arb(dat,offset,K,eta)
 
 % Created the 23rd of April 2014 by M�lodie C.A.S Hadzic
-% Last update: 147th of March 2018 by Richard B�rner
+% Last update: 12th of March 2019 by M�lodie Hadzic
+% >> obligates to give offset, K and eta values in input
 %
-% Comments adapted for Boerner et al 2017
+% update: 147th of March 2018 by Richard B�rner
+% >> Comments adapted for Boerner et al 2017
 
-dat = varargin{1};
-
-if size(varargin,2)==3
-    K = varargin{2};
-    eta = varargin{3};
-elseif size(varargin,2)==2
-    eta = varargin{2};
-    K = 57.7;
-else
-    eta = 0.95;
-    K = 57.7;
-end
-
-offset = 0; % offset implemented in camera noise model. Do not change!
-
-dat = offset + dat*eta*K; % offset implemented in camera noise model. Do not change!
+dat = offset + dat*eta*K; 
```

Comment:
@mca-sh this generates a conflict with the following code in `updateFields.m`, since these lines expect only a single input argument
```
p.totInt = phtn2arb(p.totInt);
p.totInt_width = phtn2arb(p.totInt_width);
p.bgInt_don = phtn2arb(p.bgInt_don);
p.bgInt_acc = phtn2arb(p.bgInt_acc);
```

Reply:
Agreed, the function included default values for different number of input parameters.

Reply:
This should be fixed by now.

Reply:
Ok